deps/cjs-module-lexer/include/cjs-module-lexer.h

```diff
@@ -0,0 +1,102 @@
+#ifndef __CJS_MODULE_LEXER_H__
+#define __CJS_MODULE_LEXER_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+struct Slice {
+  const uint16_t* start;
+  const uint16_t* end;
+};
+typedef struct Slice Slice;
+
+struct StarExportBinding {
+  const uint16_t* specifier_start;
+  const uint16_t* specifier_end;
+  const uint16_t* id_start;
+  const uint16_t* id_end;
+};
+typedef struct StarExportBinding StarExportBinding;
+
+void bail (uint32_t err);
+
+bool parseCJS (uint16_t* source, uint32_t sourceLen, void (*addExport)(const uint16_t*, const uint16_t*), void (*addReexport)(const uint16_t*, const uint16_t*));
```

Comment:
Suggestion and a request...

Suggestion: I would add typedefs for the addExport and addReexport callback functions and reference the typedefs here. It would make this a lot more readable.

Request: This comes in later on, but I would add a third `void* ` context argument to this and pass it to both the `addExport` and `addReexport` functions when called. Doing so will allow you to avoid the global state variables used later. Specifically, something like:

```c
typedef void (*ExportFunction)(const uint16_t*, const uint16_t*, void* userdata);

bool parseCJS (uint16_t* source,
               uint32_t sourceLen,
               ExportFunction addExport,
               ExportFunction addReexport,
               void* userdata);
```